torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2103,6 +2121,11 @@ const int& ProcessGroupNCCL::globalRank() const {
   return globalRank;
 }
 
+const c10::intrusive_ptr<Store>& ProcessGroupNCCL::globalStore() const {
+  static c10::intrusive_ptr<Store> globalStore = store_;
```

Comment:
Avoid using `static` if not necessary -- it gets messy in multi-thread scenario.
The original way of being a member variable of ProcessGroupNCCL is better.

Reply:
This is actually following the implementation of `globalRank()`.

Reply:
Sharing the same store is indeed expected because we want the default PG to detect signals from other PGs @d4l3k

Reply:
I am referring to a case of 1 thread per rank instead of 1 process per rank.
In that case, using the same store across threads (ranks) would be probably wrong.